.github/RELEASE_TEMPLATES/plugin_release_checklist.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Plugin Release {{ env.RELEASE_NAME }}
+labels: release
+---
+## Requirements
+- [ ] ⚠️ Ensure the release process has succeeded before proceeding
+
+## GCS
+- [ ] Run the `update_gcs_latest.sh` scripts in the {{ env.PROJECT_NAME }} projects to update GCS with the latest version number
+
+## Github
+- [ ] Update [CHANGELOG.md]({{ env.CHANGELOG_URL }})
+- [ ] Update [README.md]({{ env.README_URL }})
+- [ ] Update [CONTRIBUTING.md](https://github.com/GoogleContainerTools/jib/blob/master/CONTRIBUTING.md)
+- [ ] Complete [Release]({{ env.RELEASE_DRAFT }})
+- [ ] Merge PR({{ env.RELEASE_PR }})
+- [ ] Update the current [milestone](https://github.com/GoogleContainerTools/jib/milestones), roll over any incomplete issues to next milestone, mention on all fixed issues that latest release has addressed issue
+
```

Comment:
very nit: maybe we can move `mention on all fixed issues that latest release has addressed issue` to a different bullet or to the `Announce` section.